tools/setup.py

```diff
@@ -0,0 +1,55 @@
+#!/bin/python3
+# Copyright 2023 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import date
+from glob import glob
+from setuptools import setup
+import os
+
+version = '0.0.1.dev' + date.today().strftime("%Y%m%d")
+project_name = 'gematria'
+
+SO_DIRECTORIES = ['./pybind11_abseil', './gematria']
+
+SHARED_OBJECTS = []
+
+# Add all shared objects in the specified directories as only the ones
+# relevant to the python installation have been placed there by Bazel.
+for so_directory in SO_DIRECTORIES:
+    SHARED_OBJECTS += glob(os.path.join(so_directory, '**', '*.so'),
+                           recursive=True)
```

Comment:
It would be nice if we could get this from the Bazel rule. This should be doable with `genrule` type rule (it provides access to the list of input files, which we should be able to write to a text file as part of the build process).

Reply:
Thanks for the suggestion! I was able to get everything to work with a bazel `genrule` and it seemed to make everything a lot less hacky, especially since I was able to move the creation of the `pybind11_abseil/status.so` symlink to the build system, which should keep things more clean.